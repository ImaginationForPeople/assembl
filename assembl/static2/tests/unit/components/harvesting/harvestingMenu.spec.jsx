import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import HarvestingMenu from '../../../../js/app/components/harvesting/harvestingMenu';
import * as fakeData from './fakeData';
import '../../../helpers/setupTranslations';

describe('harvestingMenu component', () => {
  beforeAll(() => {
    window.getSelection = () => ({
      removeAllRanges: () => {}
    });
  });
  it('should match harvestingMenu snapshot', () => {
    const { extracts } = fakeData;
    const props = {
      extracts: extracts,
      postId: '1234',
      isHarvesting: true,
      harvestingAnchorPosition: { x: 100, y: 200 }
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<HarvestingMenu {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});