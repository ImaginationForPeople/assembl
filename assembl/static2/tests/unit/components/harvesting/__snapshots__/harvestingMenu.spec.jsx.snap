// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`harvestingMenu component should match harvestingMenu snapshot 1`] = `
<div
  className="harvesting-container"
>
  <Connect(Apollo(Apollo(Apollo(Apollo(Component)))))
    cancelHarvesting={undefined}
    displayHarvestingBox={undefined}
    extract={
      Object {
        "body": "Hello world!",
        "creationDate": "2018-03-29T16:28:27.324276+00:00",
        "creator": Object {
          "displayName": "John Doe",
          "id": "1223456",
          "userId": 31,
        },
        "extractAction": null,
        "extractNature": null,
        "id": "987643",
        "important": false,
        "textFragmentIdentifiers": Array [
          Object {
            "offsetEnd": 988,
            "offsetStart": 973,
            "xpathEnd": "//div[@id='message-body-local:Content/3059']/",
            "xpathStart": "//div[@id='message-body-local:Content/3059']/",
          },
        ],
      }
    }
    harvestingBoxPosition={null}
    isAuthorAccountDeleted={undefined}
    postId="1234"
    refetchPost={undefined}
    setHarvestingBoxDisplay={undefined}
  />
  <Connect(Apollo(Apollo(Apollo(Apollo(Component)))))
    cancelHarvesting={undefined}
    displayHarvestingBox={undefined}
    extract={
      Object {
        "body": "Hello everybody!",
        "creationDate": "2018-04-29T16:28:27.324276+00:00",
        "creator": Object {
          "displayName": "Alice Thomas",
          "id": "78965",
          "userId": 29,
        },
        "extractAction": null,
        "extractNature": null,
        "id": "75432",
        "important": false,
        "textFragmentIdentifiers": Array [
          Object {
            "offsetEnd": 123,
            "offsetStart": 22,
            "xpathEnd": "//div[@id='message-body-local:Content/3050']/",
            "xpathStart": "//div[@id='message-body-local:Content/3050']/",
          },
        ],
      }
    }
    harvestingBoxPosition={null}
    isAuthorAccountDeleted={undefined}
    postId="1234"
    refetchPost={undefined}
    setHarvestingBoxDisplay={undefined}
  />
</div>
`;
